### ISSUE TITLE
Comedy Central Broken - New URL for m3u8?

### ISSUE THREAD

**Issue by kucksdorfs** [yt-dlp/yt-dlp#8722]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

United States/Wisconsin

### Provide a description that is worded well enough to be understood

I believe the URL for the m3u8 files may have changed for Comedy Central (any they may be single files instead of a playlist) Below is last night's episode of The Daily Show (Monday night's worked fine). Opening the Network tab in the Developer tools does give me an m3u8 file, and that downloads successfully (I haven't tested all 4, but the first one is the length of the video provided from the web browser player).

https://www.cc.com/episodes/1pc7nr/the-daily-show-december-5-2023-season-28-ep-116
     https://topaz.viacomcbs.digital/h/a/dG9wYXoxYDQ3MjYwNmMwLWNjY2QtMTFlZC1hZDhlLTBlNDBjZjJmYzI4NWA2OTJmZDFkZmNjOTdmN2UzMjE5MzY3MGJjZmE5Y2JlYWY5OGZjYmY0YHI2akVvUHJINFNaSHFxNnJKOUFkRUZFeGxtTXRNel82VDBJN0xVWUVOcTA/master.m3u8?mgid=mgid:arc:episode:comedycentral.com:472606c0-cccd-11ed-ad8e-0e40cf2fc285&ts=1701854196-0&pthash=692fd1dfcc97f7e32193670bcfa9cbeaf98fcbf4&hdnts=exp=1701872554~acl=/h/a/dG9wYXoxYDQ3MjYwNmMwLWNjY2QtMTFlZC1hZDhlLTBlNDBjZjJmYzI4NWA2OTJmZDFkZmNjOTdmN2UzMjE5MzY3MGJjZmE5Y2JlYWY5OGZjYmY0YHI2akVvUHJINFNaSHFxNnJKOUFkRUZFeGxtTXRNel82VDBJN0xVWUVOcTA/*~hmac=793604cbfb66fd4ffa6b62ef1da823403d72eb3ee241ad31318ffbf2f2423b85&CMCD=mtp%3D500%2Cot%3Dm%2Csf%3Dh%2Csid%3D%223511f061-09ec-4b7c-9dc5-784e455683e0%22%2Csu
     https://topaz.viacomcbs.digital/h/a/dG9wYXoxYDQ3MjYwNmMwLWNjY2QtMTFlZC1hZDhlLTBlNDBjZjJmYzI4NWA2OTJmZDFkZmNjOTdmN2UzMjE5MzY3MGJjZmE5Y2JlYWY5OGZjYmY0YHI2akVvUHJINFNaSHFxNnJKOUFkRUZFeGxtTXRNel82VDBJN0xVWUVOcTA/hdntl=exp=1701958655~acl=%2f*~data=hdntl~hmac=4f41f1548f0379efea466f4bba895180340c1d80fae2a1dd17541feb12f7b9f6/index-f1-v1.m3u8?CMCD=mtp%3D500%2Cot%3Dm%2Csf%3Dh%2Csid%3D%223511f061-09ec-4b7c-9dc5-784e455683e0%22%2Csu
     https://topaz.viacomcbs.digital/h/a/dG9wYXoxYDQ3MjYwNmMwLWNjY2QtMTFlZC1hZDhlLTBlNDBjZjJmYzI4NWA2OTJmZDFkZmNjOTdmN2UzMjE5MzY3MGJjZmE5Y2JlYWY5OGZjYmY0YHI2akVvUHJINFNaSHFxNnJKOUFkRUZFeGxtTXRNel82VDBJN0xVWUVOcTA/hdntl=exp=1701958655~acl=%2f*~data=hdntl~hmac=4f41f1548f0379efea466f4bba895180340c1d80fae2a1dd17541feb12f7b9f6/index-f8-a1.m3u8?CMCD=mtp%3D500%2Cot%3Dm%2Csf%3Dh%2Csid%3D%223511f061-09ec-4b7c-9dc5-784e455683e0%22%2Csu
     https://topaz.viacomcbs.digital/h/a/dG9wYXoxYDQ3MjYwNmMwLWNjY2QtMTFlZC1hZDhlLTBlNDBjZjJmYzI4NWA2OTJmZDFkZmNjOTdmN2UzMjE5MzY3MGJjZmE5Y2JlYWY5OGZjYmY0YHI2akVvUHJINFNaSHFxNnJKOUFkRUZFeGxtTXRNel82VDBJN0xVWUVOcTA/hdntl=exp=1701958655~acl=%2f*~data=hdntl~hmac=4f41f1548f0379efea466f4bba895180340c1d80fae2a1dd17541feb12f7b9f6/index-f7-v1.m3u8?CMCD=mtp%3D61400%2Cot%3Dm%2Csf%3Dh%2Csid%3D%223511f061-09ec-4b7c-9dc5-784e455683e0%22%2Csu

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
>$ yt-dlp -vU https://www.cc.com/episodes/tpg24h/star-trek-very-short-treks-skin-a-cat-season-1-ep-1
[debug] Command-line config: ['-vU', 'https://www.cc.com/episodes/tpg24h/star-trek-very-short-treks-skin-a-cat-season-1-ep-1']
[debug] User config "/home/stephen/.config/yt-dlp/config": ['-f', '[ext=mp4][height<=720]/[ext=mp4][height<=1080]/[ext=mp4][height<=720]+bestaudio/[ext=mp4][height<=1080]+bestaudio/best', '--write-sub', '--sub-lang', 'en', '--extractor-args', 'twitch:subs=False']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.11.16 from yt-dlp/yt-dlp [24f827875]
[debug] Python 3.11.6 (CPython x86_64 64bit) - Linux-6.1.64-1-lts-x86_64-with-glibc2.38 (OpenSSL 3.1.4 24 Oct 2023, glibc 2.38)
[debug] exe versions: ffmpeg 6.1 (setts), ffprobe 6.1, phantomjs broken, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.12.0, certifi-2023.11.17, sqlite3-3.44.2, websockets-10.4
[debug] Proxy map: {}
[debug] Request Handlers: urllib
[debug] Loaded 1901 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2023.11.16 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2023.11.16 from yt-dlp/yt-dlp)
[ComedyCentral] Extracting URL: https://www.cc.com/episodes/tpg24h/star-trek-very-short-treks-skin-a-cat-season-1-ep-1
[ComedyCentral] star-trek-very-short-treks-skin-a-cat-season-1-ep-1: Downloading webpage
[ComedyCentral] 41738d55-83e9-11ee-ad8e-0e40cf2fc285: Downloading info
[ComedyCentral] 41738d55-83e9-11ee-ad8e-0e40cf2fc285: Extracting information
[ComedyCentral] 41738d55-83e9-11ee-ad8e-0e40cf2fc285: Downloading video urls
ERROR: [ComedyCentral] tpg24h: ComedyCentral returned error: not_found -
      Sorry, this video is not found or no longer available due to date or rights restrictions.

  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 715, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/mtv.py", line 323, in _real_extract
    videos_info = self._get_videos_info(mgid, url=url)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/mtv.py", line 227, in _get_videos_info
    return self._get_videos_info_from_url(info_url, video_id, use_hls)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/mtv.py", line 239, in _get_videos_info_from_url
    info = self._get_video_info(item, use_hls)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/mtv.py", line 150, in _get_video_info
    raise ExtractorError(message, expected=True)
```

**Comment by ParadoxGBB** [comment]
Don't know if this helps, but now I see three m3u8 references per episode. The first is a shorter one that stops before the final segment (usually just prior to the interview), the second is a lower-quality full episode that's video only, and the last is the audio of the second. Now the ads are built in --- previously I could grab the ~5 total show segments via yt-dlp and stitch them together and it was ad-free. The last two don't seem to report correctly via --list-formats.

```

using --list-formats:

"https://topaz.dai.viacomcbs.digital/h/a/dG9wYXoxYDQ3MjYwNmMwLWNjY2QtMTFlZC1hZDhlLTBlNDBjZjJmYzI4NWA4YWY4ODdjMmJmYmI5MjI5ZDdiYTIwZTg0NWRjZDBhZTAxN2ExZDk1YDBUU000dERYbTg0X28zU0liTWEzRWoySFlvS2xZSFY3TUsyQVVqZWxlSkk/master.m3u8?hdnea=st=1702334932~exp=1702338532~acl=/h/a/dG9wYXoxYDQ3MjYwNmMwLWNjY2QtMTFlZC1hZDhlLTBlNDBjZjJmYzI4NWA4YWY4ODdjMmJmYmI5MjI5ZDdiYTIwZTg0NWRjZDBhZTAxN2ExZDk1YDBUU000dERYbTg0X28zU0liTWEzRWoySFlvS2xZSFY3TUsyQVVqZWxlSkk/*~id=9a2757da-c954-439a-b0d0-df3140b45a1e~hmac=8db2e91cf287b3f435835e747cb6935212ea499446b1f648af32cda00449e650&originpath=/ondemand/hls/content/2566854/vid/mgid:arc:episode:comedycentral.com:472606c0-cccd-11ed-ad8e-0e40cf2fc285/DLS/streams/c1b08b17-2ec0-4a72-bbfa-0cb73de75105/master.m3u8&CMCD=mtp%3D500%2Cot%3Dm%2Csf%3Dh%2Csid%3D%221556951f-815b-435c-80ca-2aa213c504a5%22%2Csu"

ID                      EXT RESOLUTION │   FILESIZE   TBR PROTO │ VCODEC        VBR ACODEC     MORE INFO
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
audio0-American_English mp4 audio only │                  m3u8  │ audio only        unknown    [en] American English
281                     mp4 384x216    │ ~ 58.67MiB  281k m3u8  │ avc1.4d401e  281k video only
406                     mp4 512x288    │ ~ 84.81MiB  407k m3u8  │ avc1.4d401f  407k video only
700                     mp4 640x360    │ ~146.17MiB  701k m3u8  │ avc1.4d401f  701k video only
877                     mp4 768x432    │ ~182.98MiB  877k m3u8  │ avc1.4d401f  877k video only
1295                    mp4 960x540    │ ~270.34MiB 1296k m3u8  │ avc1.4d401f 1296k video only
1962                    mp4 1280x720   │ ~409.35MiB 1962k m3u8  │ avc1.640028 1962k video only
4163                    mp4 1920x1080  │ ~868.51MiB 4163k m3u8  │ avc1.640028 4163k video only

"https://topaz.dai.viacomcbs.digital/ondemand/hls/content/2566854/vid/mgid:arc:episode:comedycentral.com:472606c0-cccd-11ed-ad8e-0e40cf2fc285/DLS/streams/c1b08b17-2ec0-4a72-bbfa-0cb73de75105/media/281dd385dae3d96cede76042fd7753ab.m3u8?hdntl=exp=1702421933~acl=%2f*~id=9a2757da-c954-439a-b0d0-df3140b45a1e~data=hdntl~hmac=b3415f0511bdcb0397eaf2d2ca418f82e2bd079ffcc9bab0c19b1bdc0dab4943&CMCD=mtp%3D500%2Cot%3Dm%2Csf%3Dh%2Csid%3D%221556951f-815b-435c-80ca-2aa213c504a5%22%2Csu"

ID EXT RESOLUTION │ PROTO │ VCODEC  ACODEC
───────────────────────────────────────────
0  mp4 unknown    │ m3u8  │ unknown unknown

"https://topaz.dai.viacomcbs.digital/ondemand/hls/content/2566854/vid/mgid:arc:episode:comedycentral.com:472606c0-cccd-11ed-ad8e-0e40cf2fc285/DLS/streams/c1b08b17-2ec0-4a72-bbfa-0cb73de75105/media/7837cc908156191f9907af444be1df55.m3u8?hdntl=exp=1702421933~acl=%2f*~id=9a2757da-c954-439a-b0d0-df3140b45a1e~data=hdntl~hmac=b3415f0511bdcb0397eaf2d2ca418f82e2bd079ffcc9bab0c19b1bdc0dab4943&CMCD=mtp%3D500%2Cot%3Dm%2Csf%3Dh%2Csid%3D%221556951f-815b-435c-80ca-2aa213c504a5%22%2Csu"

ID EXT RESOLUTION │ PROTO │ VCODEC  ACODEC
───────────────────────────────────────────
0  mp4 unknown    │ m3u8  │ unknown unknown
```

**Comment by ParadoxGBB** [comment]
Hello all,

In honor of Jon Stewart’s upcoming return to The Daily Show, I am happy to report that I was able to debug and fix this on my machine. I’m very poor at Python however so my guess is that anyone looking at the code will want to suggest modifications. Although the subtitles are in the HLS manifest so far I haven’t figured out how to extract them yet. Right now I get the following warning by the tool and subtitle base64 junk at the bottom of the video:

`Ignoring subtitle tracks found in the HLS manifest; if any subtitle tracks are missing, please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U`

There are a lot of overloaded and modified versions of the same functions so I might have broken other MTV properties like South Park or Nickelodeon. There also might be a better way to extract and promote more information in the video information --- right now the title is the end of the episode url and I guess we could promote the episode and season from that as well.

If someone in the community could help me get this over the finish line, I’d be much obliged.

Basically,

1. You can extract the video_id as before.
2. You can use that to calculate and extract a new location for the JSON feed for the episode.
3. That JSON has a field that has the full URL for the m3u8 playlist. The token for authorization is built right into the url.

The reference URL I used was:

https://www.cc.com/episodes/fppd9b/the-daily-show-december-14-2023-taraji-p-henson-season-28-ep-122

To apply the patch and help test:

1. Replace line 6 in the comedycentral.py extractor to point to the new feed: 
          `_FEED_URL = 'https://topaz.viacomcbs.digital/topaz/api/'`
2. Replace mtv.py with https://pastebin.com/izkP3K19.

This appears to work for newer content --- it does not work for things published before about 12/8/2023.

**Comment by Hamuko** [comment]
> To apply the patch and help test:
> 
>     1. Replace line 6 in the comedycentral.py extractor to point to the new feed:
>        `_FEED_URL = 'https://topaz.viacomcbs.digital/topaz/api/'`
> 
>     2. Replace mtv.py with https://pastebin.com/izkP3K19.

Seems to work quite beautifully for TDS S29E01.

**Comment by atpage** [comment]
The proposed patch worked for me a couple of times, but seems broken again.

    An extractor error has occurred. (caused by KeyError('stitchedstream'))

**Comment by ParadoxGBB** [comment]
> The proposed patch worked for me a couple of times, but seems broken again.
> 
> ```
> An extractor error has occurred. (caused by KeyError('stitchedstream'))
> ```

Do you have a reference URI? Worked fine with me for last night's TDS in the States.

**Comment by atpage** [comment]
> > The proposed patch worked for me a couple of times, but seems broken again.
> > ```
> > An extractor error has occurred. (caused by KeyError('stitchedstream'))
> > ```
> 
> Do you have a reference URI? Worked fine with me for last night's TDS in the States.

Aha!  "in the States" was the problem.  VPN issue, not yt-dlp issue.  Thanks.

**Comment by pishposhmcgee** [comment]
> Although the subtitles are in the HLS manifest so far I haven’t figured out how to extract them yet. Right now I get the following warning by the tool and subtitle base64 junk at the bottom of the video:
> 
> `Ignoring subtitle tracks found in the HLS manifest; if any subtitle tracks are missing, please report this issue on https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using yt-dlp -U`

I am using the linked modifications and made my own changes to pick up HLS subtitles:

I replaced the following section (starting at line 232):
`        formats = []
        if feed['stitchedstream']['manifesttype'] == 'hls':
            formats.extend(self._extract_m3u8_formats(
                playlist, video_id, ext='mp4', entry_protocol='m3u8_native',
                m3u8_id='hls', fatal=True))`
With
`       if feed['stitchedstream']['manifesttype'] == 'hls':
            formats, subtitles = self._extract_m3u8_formats_and_subtitles(playlist, video_id, ext='mp4', entry_protocol='m3u8_native',
                m3u8_id='hls', fatal=True)
        else:
            formats=[]
            subtitles=[]`

Then replaced
`
            'formats': formats,
            'id': video_id,`
With
`
            'formats': formats,
            'subtitles': subtitles,
            'id': video_id,`

That picks up the subs embedded in the HLS stream.

I know it was mentioned above, but playlists are still broken (e.g. https://www.cc.com/playlists/dm1h37/the-daily-show-extended-interviews/50p5ws only returns one entry). More importantly, it would be a big problem to just substitute in this new extractor code if it breaks any video from before 2024.

**Comment by pishposhmcgee** [comment]
With some further investigation it looks like the existing `mtv.py` extractor is broken for at least mtv.com. changing the `_FEED_URL` string in the MTVIE section also fixes MTV. The same is the case for TVLand. It looks like any of the sites that depend on the MTV infrastructure are being shifted over to the newer JSON infrastructure, and they all seem to use the same `_FEED_URL` endpoint.

**Comment by ParadoxGBB** [comment]
@pishposhmcgee , thanks for the subtitles and testing legwork. I've been meaning to spend some time to understand the testing framework a bit more before proceeding with this.

Oddly the videos that are being extracted do still have a junk subtitle stream in the mp4. Is anyone else seeing this? It appears to be extracted somehow downstream and the m3u8 doesn't appear to have multiple references to it so I'm not sure yet where it's coming from.

Regarding playlists, I don't know how it worked before, but functionally from what I can see there's nothing that separates pages like /episodes, /videos, and /playlists, both serve up the current video and then on completion redirects to the next one on the queue one is just presented more explicitly as a theme whereas other one serves up additional pre-selected related content. Typically I would think a traditional playlist would have multiple media references in the feed (here, mica.json) and that doesn't seem to be the case. I welcome any corrections if my understanding is incorrect.

Previously, CC's extractor used to grab each chapter/segment and I needed to stitch them together explicitly. Now the feed has a parent section that references the whole thing sans ads, but in the few cases I've tested the feed content section only contains 1 media file and the playlist information are urls passed by some other mechanism that in turn could be iterated and have the M3u8 extracted, but I don't think most playlists are extracted this way.

I don't have a timeframe yet but I can message this a bit more and move the feed URL to the base class and move forward with a PR.

Regarding backwards compatibility, I'm not sure if that's a bar that's been consistently cleared by other extractors as most media references tend to rotate in these streaming services (or consolidated in otther services ;-)). But I can look at the error to see if forking logic could try to do a good faith attempt at older content.

**Comment by Wowfunhappy** [comment]
Hello! The posted patch by @ParadoxGBB has been working with beautiful consistency for more than a month. Is there any chance of getting this upstreamed in the near-ish future? Subtitle support can presumably come later, right?

**Comment by ParadoxGBB** [comment]
Hey guys,

Some updates:

1) The subtitle junk is because the feed I'm using promotes a thumbnail in WebVTT format, which I had no idea existed until now (https://developer.bitmovin.com/playback/docs/webvtt-based-thumbnails). Does any player out there actually understand this? I removed that logic and so no thumbnail for individual episodes. Oddly enough the jpg thumbnail (not the video thumbnail, the playlist thumbnail) is still extracted for videos under the /playlist/ url structure and I'm not sure how that gets extracted / promoted.

2) With playlists I confirmed that the feed always is presented the same, as a single video. The list of videos (for example, TDS's extended interviews) are listed in embedded json in the webpage but my understanding is that it's not used for most extractors, the feed URL is preferred and would extract everything as a playlist, even from /episodes if it was the same. However that said 1)'s promotion of the thumbnail might indicate I don't understand this fully.

3) I've promoted season / episode number based on the url, and am grabbing the correct (non-thumbnail) subtitles now thanks to the help here (thanks again!).

4) The code does appear to be backwards compatible with older content I think (clips of South Park) but not compatible with older content that is now behind provider authentication. For videos like those I'm getting the same StichedStream error mentioned above. I'm looking through the code to see if there's a common way can pass along the token from --ap-username and --ap-password auth --- it appears to be passed into the feed request as a header.

5) I've reviewed some of the documents around running tests and will run / fix / spread the feed logic to all of the classes I see that inherits from this base MTV class.

I'll take direction and nudges for items 1/2, but that seems good enough to me. When I have this at an acceptable functional level for 4/5 I'll send a PR.

**Comment by Wowfunhappy** [comment]
Hello again! It has been a couple of months, so I wanted to check in. 

I have been using the posted patch since February, and it really has been working beautifully! The only hiccup is that I always have to switch to a different virtual environment every time I want to download from Comedy Central. If the patch was made a part of the main yt-dlp package, things would be a lot easier!

I understand that the current patch isn't perfect. However, it's obviously superior to the current upstream extractor, which doesn't work at all! Therefor, I would really encourage merging this patch in its current state if possible.

This isn't meant to rush anyone's work—I'm mostly just happy to have a way to download from Comedy Central—but I can't help feeling that the current situation is non-optimal for everyone! Regardless of what you decide, thank you for the wonderful fix!

**Comment by kucksdorfs** [comment]
@Wowfunhappy I just reported the bug, I haven't done anything to help fix it. Just so you know, if you go directly for the m3u8 file (dev tools -> network tab) it does get the subtitle file. 

`
yt-dlp "https://topaz.viacomcbs.digital/h/a/dG9wYXoxYDA5YmEwODE4LWViYWMtMTFlZS1hZDhlLTBlNDBjZjJmYzI4NWAzOTQ1YWJjOWZlYzg4NDY5Zjc3ODVhNThiYjlhNDI2NTdhNmUwMGViYFNHaXRkOWN2N3czaXI1THY4U2oyZ1dORV9oM0pvWjhMZjAwQnJGbkhXdFU/master.m3u8?mgid=mgid:arc:episode:comedycentral.com:09ba0818-ebac-11ee-ad8e-0e40cf2fc285&ts=1717578603-0&pthash=3945abc9fec88469f7785a58bb9a42657a6e00eb&hdnts=exp=1717587844~acl=/h/a/dG9wYXoxYDA5YmEwODE4LWViYWMtMTFlZS1hZDhlLTBlNDBjZjJmYzI4NWAzOTQ1YWJjOWZlYzg4NDY5Zjc3ODVhNThiYjlhNDI2NTdhNmUwMGViYFNHaXRkOWN2N3czaXI1THY4U2oyZ1dORV9oM0pvWjhMZjAwQnJGbkhXdFU/*~hmac=4b06f8bca2b8183daf89f9af119f066c0f74ff1def447f53447b0a5f4bce33c4&CMCD=mtp%3D500%2Cot%3Dm%2Csf%3Dh%2Csid%3D%22450b33b3-c735-4b0b-a137-65ebf15785a1%22%2Csu" -o 202460604_Episode.mp4 --write-sub
`

**Comment by ParadoxGBB** [comment]
All,

First off, apologies for the silence from me --- as I indicated before I really hate coding in Python so I haven't been too motivated to fix this beyond what I have, but I do acknowledge that it could have been more compatible with other yt-dlp features (playlist, auth). I'm very sorry to those that's been waiting on me.

However, starting today I think this fix may now be moot --- now cc.com's show series pages are redirecting to paramount plus's series landing pages and it looks like all of the content as now behind a subscription paywall there. I'd assume other properties like MTV and TVLand will follow suit soon if not already.

Because I'm curious about this I was poking around a bit today and found that cable subscribers might apparently still have a video on demand source if you have a valid login (for example, I was looking at https://tv.verizon.com/watch/home/) so, somewhere there's probably still a M3u8 published on this behind authentication. However they also have code that detects and redirects you if you start the browser debugging tools (sneaky!) so getting at it might take some doing.

All of that still means that as a publishing source, comedycentral.com / cc.com may be no more. If I'm mistaken I'd love to be corrected on this.

**Comment by pishposhmcgee** [comment]
RIP

**Comment by kucksdorfs** [comment]
Suppose we can close dieto DRM...

**Comment by Wowfunhappy** [comment]
Just checking—do we actually know that the new site is DRM'd, as opposed to just requiring a login?

**Comment by ParadoxGBB** [comment]
> Just checking—do we actually know that the new site is DRM'd, as opposed to just requiring a login?

Folks should weigh in if my understanding is incorrect of course --- 

The content for paramountplus.com is DRM'd normally --- as well as other Viacom properties like CBS.com (you can search for Colbert for issues here to get some background on that).

Cable channel sites, like TBS, History, Comedy Central, USA, TNT, MTV, etc, used to allow some of their content for "free" but older content required authentication like via their plus offerings or via a cable subscriber. Now that ComedyCentral.com's usefulness seems to have gone away here, that option seems harder to utilize.

That doesn't mean that this content isn't available through legal non-DRM'd channels, necessarily (see my Verizon observation above) but it means as an extractor comedycentral.com might have its days numbered at this point.

**Comment by bashonly** [comment]
Keeping this open for tracking purposes until the extractor is removed (or fixed?)
